extension/src/test/suite/project_findreplace.test.ts

```diff
@@ -0,0 +1,232 @@
+import * as path from 'path';
+import { OpenProjectFile } from '../../project/openProject';
+import * as vscode from 'vscode';
+import { ProjectTreeProvider } from '../../project/projectTree';
+
+import { findInProject, FindInProject } from '../../project/findReplace/findInProject';
+import { SearchOption } from '../../project/findReplace/options';
+import { applyReplacementInFile } from '../../project/findReplace/applyReplacement';
+
+import * as os from 'os';
+import { ImportMock } from 'ts-mock-imports';
+import { FileNode, SearchTreeProvider } from '../../project/findReplace/searchTree';
+import { ReadonlyDocument } from "../../project/readOnlyDocument";
+import * as ut from '../../utils';
+import * as opt from '../../project/findReplace/options';
+import * as fs from 'fs';
+
+import { clearSearchResultWithError, setIsSearching, getWarnIsSearching } from "../../project/findReplace/clearResults";
+import { openSearchResult } from "../../project/findReplace/openSearchResult";
+import { FindingNode } from '../../project/findReplace/searchTree';
+import { replaceInProject } from '../../project/findReplace/replaceInProject';
+import { findInFile, grantExePermission } from '../../project/findReplace/ripGrep';
+import { saveOpenDoc2Tmp } from '../../utils';
+import { detectEncoding } from '../../project/findReplace/encoding';
+
+var expect = require('chai').expect;
+suite("Project find and replace in project Tests", function () {
+	var filenode = new FileNode;
+	let prefixpath = __filename + "/../../../../extension/src/test/SourceFile/test_case/";
+	let projpath = path.join(prefixpath + "project_test_file.prj");
+	let ret = vscode.Uri.file(projpath);
+	let rootNode= OpenProjectFile(ret);
+
+	var searchobj = new SearchOption();
+	searchobj.keyword = "-layer";
+	searchobj.completed = true;
+
+	var mock_value = "bbbbbbb";
+
+	ProjectTreeProvider.instance().updateData(rootNode);
+		
+	let closeParenStyleStub;
+	suiteTeardown( async ()=>{
+		closeParenStyleStub.restore();
+	});
+	suiteSetup(async () => {
+		closeParenStyleStub = ImportMock.mockFunction(os, 'arch', 'x64');
+		closeParenStyleStub = ImportMock.mockFunction(opt, 'getSearchOption', searchobj);
+		closeParenStyleStub = ImportMock.mockFunction(vscode.commands, 'executeCommand', mock_value);
+		closeParenStyleStub = ImportMock.mockFunction(opt, 'getString', "remove -layer");
+		closeParenStyleStub = ImportMock.mockFunction(os, 'type', 'Darwin');
+		closeParenStyleStub = ImportMock.mockFunction(fs, 'accessSync', null);
+		closeParenStyleStub = ImportMock.mockFunction(fs, 'chmodSync', null);
```

Comment:
please give each stub an identical name, and restore them all in suiteTeardown

Reply:
Got it.